### PR TITLE
STL 631 - Transaction Execution Results

### PR DESCRIPTION
**Pull request by aludvik**
This PR updates the context manager, executor, and chain controller to enable passing receipts from a transaction processor up to chain observers. It then makes the receipt store observe chain updates and save receipts and enables message handling to allow querying for receipts.

### CODE REVIEW

**Comment by aludvik** [comment]
Added comments to commits

File: validator/sawtooth_validator/journal/chain.py [hunk 1]
```diff
@@ -797,7 +797,7 @@ def _set_genesis(self, block):
                            "genesis block: %s", block)
 
     def _make_receipts(self, results):
-        receipts = {}
+        receipts = []
```
Comment (agunde406) [inline validator/sawtooth_validator/journal/chain.py]: This change should be squashed with the previous commit

File: validator/tests/test_context_manager/tests.py [hunk 2]
```diff
@@ -183,6 +184,40 @@ def _create_txn_inputs_outputs(self, start=None):
...
+        sets = {addr1: b'1'}
+        events = [Event(
+            event_type="test",
+            attributes=[Event.Attribute(key="test", value="test")],
+            data="test".encode())]
+        deletes = {addr2: None}
+        data = [("test", "test".encode())]
+
+        self.context_manager.set(context_id, [sets])
+        for event in events:
```
[the remark this answers is not in the document]
Comment (aludvik) [reply validator/tests/test_context_manager/tests.py]: Updated so there is more than one.

File: validator/sawtooth_validator/journal/chain.py (hunk 1 above)
Comment (aludvik) [reply validator/sawtooth_validator/journal/chain.py]: Fixed

**Pull request merged**